2748. Number of Beautiful Pairs/solution.py

```diff
@@ -0,0 +1,31 @@
+class Solution:
+    # This takes in a list of nums and returns each beautiful pair,
+    # where a beautiful pair is when the first digit of the first
+    # num in the pair is coprime with the last digit of the second
+    # num in the pair. Two nums are coprime if their gcd is no greater
+    # than 1.
+    def countBeautifulPairs(self, nums: List[int]) -> int:
+        # the number of beautiful pairs
+        
+        # convert each int in the list to strings to save
+        # time from converting each int to a string inside the 
+        # main loop
+        
+        # iterate to the second-to-last digit of nums
+        
+            # iterate through the next nums
+            
+                # find the first char of the first digit
+                
+                # find last char of second digit (convert both to int)
+                
+                # iterate backwards from 9 to 2
```

Comment:
Two things. First, shouldn't you create a separate function for this? Secondly, you should iterate from the smaller number down to 2.

Reply:
If you put this in your GCD function, you'll also will have to iterate from the smaller number to 1. And you'll have to iterate through the smaller number down to 1 if you want to handle numbers greater than 10 in other cases! I'd generally recommend generalizing your functions.

Reply:
No this is a solution for the specific problem and you want it to be as fast as possible.